core/src/avm2/globals/xml.rs

```diff
@@ -101,3 +101,66 @@ pub fn children<'gc>(
     )
     .into())
 }
+
+pub fn attributes<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let this = this.unwrap();
+    let xml = this.as_xml_object().unwrap();
+    let attributes = if let E4XNodeKind::Element { attributes, .. } = &*xml.node().kind() {
+        // FIXME - avoid clone
+        attributes.clone()
```

Comment:
Should be doable if we just move `attributes.iter().map(|node| E4XOrXml::E4X(*node)).collect()` here, right?